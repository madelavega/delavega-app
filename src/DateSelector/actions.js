const actionPrefix = 'DATE_SELECTOR';
const genAction = (action) => `${actionPrefix}_${action}`;

export const actions = {
    SELECTED_DATE: genAction('SELECTED_DATE'),
};

export const actionCreators = {
    /**
     *
     * @param {String} date
     * @returns {{payload: {date}, type: string}}
     */
    selectDate: (date) => ({
        type   : actions.SELECTED_DATE,
        payload: { date },
    }),
};

