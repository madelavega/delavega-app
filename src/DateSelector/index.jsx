import React, { useRef, useEffect } from 'react';
import { useHistory } from 'react-router-dom';

import PropTypes from 'prop-types';
import 'delavega-lib';

function DateSelector(props) {
    let history = useHistory();
    const dropDown = useRef(),
        { selectedDate } = props;

    /**
     *
     * @param selectedValue
     */
    const onSelected = ({ detail : { selectedValue } }) => {
        props.selectDate?.(selectedValue);
        if(!!selectedValue && selectedValue !== selectedDate) {
            history.push(`/results/${selectedValue}`);
        }
    };

    /**
     *
     */
    useEffect(() => {
        const { data, defaultEmptyValue } = props,
            currentNode = dropDown.current;
        currentNode.data = data;
        currentNode.defaultEmptyValue = defaultEmptyValue;
        currentNode.value = selectedDate;
        currentNode.addEventListener('selected', onSelected);

        return () => {
            currentNode.removeEventListener('selected', onSelected);
        };
    });

    return (
        <div className="App">
            <delavega-dropdown
                ref={dropDown}
            />
        </div>
    );
}

DateSelector.propTypes = {
    data: PropTypes.arrayOf(PropTypes.shape({
        value: PropTypes.string,
        text : PropTypes.string,
    })),
    defaultEmptyValue: PropTypes.bool,
    selectDate       : PropTypes.func,
    selectedDate     : PropTypes.string,
};

export default DateSelector;