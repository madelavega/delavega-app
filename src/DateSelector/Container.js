import { actionCreators } from './actions';
import DateSelector from './index';
import { connect } from 'react-redux';
const mapStateToProps = (state) => {
    const {
        dateSelector : { data, selectedDate },

    } = state;

    return {
        data,
        selectedDate,
    };
};


const mapDispatchToProps = dispatch => {
    return {
        selectDate: (selectedDate) => dispatch(actionCreators.selectDate(selectedDate)),
    };
};

export default connect(mapStateToProps, mapDispatchToProps)(DateSelector);
