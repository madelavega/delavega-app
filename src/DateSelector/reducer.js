import { actions } from './actions';

const initialState = {
    selectedDate: null,
    dates       : [
        { value: '01-01-2021', text: '01-01-2021' },
        { value: '02-02-2021', text: '01-02-2021' },
        { value: '01-03-2021', text: '01-03-2021' },
        { value: '01-04-2021', text: '01-04-2021' },
        { value: '01-05-2021', text: '01-05-2021' },
        { value: '01-06-2021', text: '01-06-2021' },
    ],
};

const reducer = (state = initialState, action) => {
    const actionTypes = {
        [actions.SELECT_DATE]: () => {  // we won't reset on start, only on hide (using RESET_AND_HIDE)
            const { payload: { date: selectedDate } = {} } = action;
            return {
                ...state,
                selectedDate,
            };
        },
    };
    return actionTypes[action.type] ? actionTypes[action.type]() : { ...state };
};

export default reducer;
