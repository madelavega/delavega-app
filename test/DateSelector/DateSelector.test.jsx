import React from 'react';
import renderer from 'react-test-renderer';
import DateSelector from '../../src/DateSelector';

it('[DateSelector] renders correctly', () => {
    const tree = renderer
        .create(<DateSelector/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});