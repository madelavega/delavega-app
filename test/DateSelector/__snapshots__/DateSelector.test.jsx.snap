// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[DateSelector] renders correctly 1`] = `
<div
  className="App"
>
  <delavega-dropdown />
</div>
`;
